### PR TITLE
Upgrade grunt-sass

### PR DESCRIPTION
**Pull request by dtothefp**
- pull grunt-sass directly from GitHub
- pull css-smart-grid from forked optimizely repo

current error:
```
>> You may not @extend an outer selector from within @media.
>> You may only @extend selectors within the same directive.
>> From "@extend .clearfix" on line 337 of website-guts/assets/css/helper-classes.scss
```

reference 
http://thesassway.com/intermediate/understanding-placeholder-selectors#limitations

these changes are confusing because of the `@extend` error. If you want to learn more read 
http://www.sitepoint.com/cross-media-query-extend-sass/
but I basically copy/pasted this code.

#### To Test

- Check all pages with changes in different browser sizes
- run `grunt build` to see if CSS is minified
- check staging https://www.optimizelystaging.com/dfoxpowell/upgrade-grunt-sass/

### CODE REVIEW

**Comment by stewsmith** [comment]
@dtothefp All the layout looks good. On /jobs the background image is not being pulled in. Same for /events -- the shadow image is not there. Do you think this is related to this PR or is just a staging issue?

File: grunt/sass.js [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
+      functions: {
+        'image-url($filename)': function(filename) {
+          var imageUrl = grunt.config.get('imageUrl');
+          var fileName = filename.getValue();
+
+          var imagePath = 'url("' + path.join(imageUrl, fileName) + '")';
+          return new sass.types.String(imagePath);
+        }
+      }
     },
```
Comment (dtothefp) [inline grunt/sass.js]: @stewsmith this is awesome, added a custom `image-url` function to Libsass by passing the `functions` option in the config

Comment (stewsmith) [inline grunt/sass.js]: That's a great fix.

**Comment by stewsmith** [comment]
LGTM

**Pull request merged**